kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/preprocess/SinkRecordFilter.java

```diff
@@ -0,0 +1,103 @@
+package com.wepay.kafka.connect.bigquery.preprocess;
+
+import com.jayway.jsonpath.*;
+import com.wepay.kafka.connect.bigquery.config.BigQuerySinkConfig;
+import com.wepay.kafka.connect.bigquery.convert.RecordConverter;
+import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.apache.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class SinkRecordFilter {
+    private BigQuerySinkConfig bigQueryConfig;
+    private RecordConverter recordConverter;
+
+    private String configType = null;
+    private String configString = null;
+    private JsonPath jsonPathCondition = null;
+    private static final Logger logger = Logger.getLogger(SinkRecordFilter.class);
+
+    public SinkRecordFilter(BigQuerySinkConfig bigQueryConfig) {
+        this.bigQueryConfig = bigQueryConfig;
+        this.recordConverter = bigQueryConfig.getRecordConverter();
+
+        if (this.bigQueryConfig.getString(this.bigQueryConfig.INCLUDE_CONDITION_CONFIG) != null) {
+            this.configType = this.bigQueryConfig.INCLUDE_CONDITION_CONFIG;
+            this.configString = this.bigQueryConfig.getString(this.bigQueryConfig.INCLUDE_CONDITION_CONFIG);
+
+        } else if (this.bigQueryConfig.getString(bigQueryConfig.EXCLUDE_CONDITION_CONFIG) != null) {
+            this.configType = this.bigQueryConfig.EXCLUDE_CONDITION_CONFIG;
+            this.configString = this.bigQueryConfig.getString(bigQueryConfig.EXCLUDE_CONDITION_CONFIG);
+            this.jsonPathCondition = JsonPath.compile(this.configString);
+        }
+
+        if (configType != null) {
+            try {
+                this.jsonPathCondition = JsonPath.compile(this.configString);
+            } catch (InvalidPathException e) {
+                throw new InvalidPathException("Json Path " + this.configString + "specified in " +
+                        configType + "config is incorrectly formatted. " +
+                        "Please refer to com.jayway.jsonpath java doc for correct use of jsonpath.");
+            }
+        }
+
+    }
+
+
+    public List getFilterFields(Map<String, Object> convertedRecord) {
+        List filterList = new ArrayList();
+
+        if (this.configType != null && convertedRecord != null) {
+            try {
+                // return a list map nodes that satify the specified condition
+                filterList = this.jsonPathCondition.read(
+                        convertedRecord,
+                        Configuration.defaultConfiguration().addOptions(
+                                Option.ALWAYS_RETURN_LIST
+                        )
+                );
+            } catch (PathNotFoundException e) {
+                logger.info("Json Path " + this.configString + " specified in " + this.configType +
```

Comment:
this should probably be trace log, otherwise it will spam the logs badly

Reply:
also, the logger can use placeholders in the log message to avoid the cost of concatenating the string if the loglevel would prevent this message from getting logged in the first place.